### ISSUE TITLE
Question about the overshooing KL divergence cost scaling.

### ISSUE THREAD

**Issue by ZhengYi0310** [Kaixhin/PlaNet#23]
Hello,
      Thanks very much for the amazing codebase ! 
      
After reviewing the codebase and the orignal PlaNet paper, I have a question about the scaling of overshooting KL divergence. at [line 171 ~ line 173 of the main.py](https://github.com/Kaixhin/PlaNet/blob/c18a37f1e684df17888e468699b01c699863ba48/main.py#L171), I notice that the observation_loss, reward_loss and one-step prediction KL divergence loss are all averaged over the chunk size and batch, which is indicated by the **.mean(dim=(0, 1))** operation. But I also notice [at line 190 of the main.py](https://github.com/Kaixhin/PlaNet/blob/c18a37f1e684df17888e468699b01c699863ba48/main.py#L190), there is a scaling of **(args.chunk_size - 1)** after the stacked multistep prediction(from d=2 to d=D) kl loss takes the **.mean(dim=(0, 1))** operation. I am wondering what's the reason behind this scaling ? Is it because that when the **overshooting_distance >= chuck_size**, then the **.mean(dim=(0, 1))** operation applied on the stacked multistep prediction kl loss first sum the kl loss up and then divide it by (1 + 2 + ... + (chuck_size - 1) = O((chunk_size - 1)^2)), therefore applying the **(args.chunk_size - 1)** will make the scaling back to O(chunk_size - 1), therefore roughly make the multistep prediction kl loss averaged over the chunk size ? If this is the case then does that mean if **overshooting_distance << chuck_size (e.g.chuck_size = 50, overshooing_distance=20)**, to achieve the same level of scaling we need to tune **overshooting_kl_beta** to be smaller than 1 ? I notice you mentioned something like this in this post [PyTorch Port](https://github.com/google-research/planet/issues/28#issuecomment-489461810), but just want to double-check.
 

 
Thanks very much for your time and patience !

**Comment by Kaixhin** [comment]
My process to replicate prioritised 1) paper 2) results 3) original code, and that's the way it ended up. But in the end the scaling is somewhat arbitrary, as the losses could interchange sum/mean, and it is a hyperparameter to be tuned for your specific problem in the end.
